src/esssans/loki/masking.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+"""
+Masking functions for the loki workflow.
+"""
+from typing import NewType, Optional
+
+import numpy as np
+import scipp as sc
+
+from ..types import (
+    BeamStopPosition,
+    BeamStopRadius,
+    MaskedData,
+    RawData,
+    RunType,
+    SampleRun,
+)
+
+DetectorLowCountsStrawMask = NewType('DetectorLowCountsStrawMask', sc.Variable)
+"""Detector low-counts straw mask"""
+DetectorBeamStopMask = NewType('DetectorBeamStopMask', sc.Variable)
+"""Detector beam stop mask"""
+DetectorTubeEdgeMask = NewType('DetectorTubeEdgeMask', sc.Variable)
+"""Detector tube edge mask"""
+
+
+def detector_straw_mask(
+    sample_straws: RawData[SampleRun],
+) -> DetectorLowCountsStrawMask:
+    return DetectorLowCountsStrawMask(
+        sample_straws.sum('pixel').data
+        < sc.scalar(sample_straws.sizes['pixel'] * 0.5, unit='counts')
+    )
+
+
+def detector_beam_stop_mask(
+    sample_straws: RawData[SampleRun],
+    beam_stop_position: BeamStopPosition,
+    beam_stop_radius: BeamStopRadius,
+) -> DetectorBeamStopMask:
+    pos = sample_straws.coords['position'] - beam_stop_position
+    pos.fields.z *= 0.0
+    return DetectorBeamStopMask((sc.norm(pos) < beam_stop_radius))
+
+
+def detector_tube_edge_mask(
+    sample_straws: RawData[SampleRun],
+) -> DetectorTubeEdgeMask:
+    other_dims = set(sample_straws.dims) - {'pixel'}
+    size = np.prod([sample_straws.sizes[dim] for dim in other_dims])
+    return DetectorTubeEdgeMask(
+        sample_straws.sum(other_dims).data < sc.scalar(size * 0.5, unit='counts')
+    )
```

Comment:
This does not look like a tube edge mask, but is masking based on counts. Why? I thought we should not masks low counts especially in SANS where low counts at large Q is expected.

Reply:
A comment should be the absolute minimum. Has this been check with the scientists?